#167619168 Create Sessions Table

ayodejiAA opened the pull request:
#### What does this PR do?
- Create migration for session table
- Add demo seed file

#### Description of Task proposed in this pull request?
-  Add session migration file
- Add Session model
- Add Session seed file

#### How should this be manually tested (Quality Assurance)?
- Check out the branch
-Add the database connection string in .env file
- Run ```npm run db:seed```

#### What are the relevant pivotal tracker stories?
- Story ID: [#167619168](https://www.pivotaltracker.com/story/show/167619168)

#### Any background context you want to add (Operations Impact)?
- Requires database connection string for it to work

#### What I have learned working on this feature:
- Use sequelize cli to generate migrations and seeds

#### Screenshots: [If you made some visual changes to the application please upload screenshots here, or remove this section]
- n/a

ayodejiAA committed:
chore-[167619168]: Create sessions table
- Create migration from session table
- Add demo seed file for session details

[Finishes #167619168]

ayodejiAA committed:
chore-[167619168]: Create sessions table
- Create migration from session table
- Add demo seed file for session details

[Finishes #167619168]

ayodejiAA committed:
chore-[167619168]: Create sessions table
- Create migration for session table
- Add demo seed file for session details

[Finishes #167619168]

dinobi commented on server/database/migrations/20190731135151-create-session.js:
```diff
@@ -0,0 +1,49 @@
+      type: Sequelize.STRING
+    },
+    userAgent: {
+      allowNull: false,
```
allow null here

ayodejiAA replied:
Done

dinobi commented on server/database/migrations/20190731135151-create-session.js:
```diff
@@ -0,0 +1,49 @@
+      type: Sequelize.DATE
+    },
+    ipAddress: {
+      allowNull: false,
```
allow null here

ayodejiAA replied:
Done

chialuka replied:
I think allow null here means that the field should be nullable, or am I wrong?

dinobi commented on server/database/models/session.js:
```diff
@@ -0,0 +1,16 @@
+      userId: DataTypes.INTEGER,
+      active: DataTypes.BOOLEAN,
+      devicePlatform: DataTypes.STRING,
+      expiresAt: DataTypes.DATE,
```
Don't sequelize have a date time type?

dinobi replied:
this is datetime, confirmed.

dinobi reviewed (changes requested):
Please add model tests

ayodejiAA commented:
> Please add model tests

Okay. On it.

ayodejiAA commented:
> Please add model tests
@dinobi  Implemented the model test

chialuka commented on server/database/migrations/20190731135151-create-session.js:
```diff
@@ -0,0 +1,49 @@
+module.exports = {
```
Please consider changing this to ES6 exports

ayodejiAA replied:
Done

ayodejiAA committed:
chore-[167619168]: Create sessions table
- Create migration from session table
- Add demo seed file for session details

[Finishes #167619168]

ayodejiAA committed:
chore-[167619168]: Create sessions table
- Create migration from session table
- Add demo seed file for session details

[Finishes #167619168]
